src/formatDate.js

```diff
@@ -8,7 +8,68 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  if (typeof date !== 'string' || date.length === 0) {
+    return '';
+  }
+
+  if (
+    !fromFormat ||
+    !toFormat ||
+    typeof fromFormat !== 'object' ||
+    typeof toFormat !== 'object' ||
+    !('length' in fromFormat) ||
+    !('length' in toFormat)
+  ) {
+    return '';
+  }
+
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+
+  if (!fromSeparator || !toSeparator) {
+    return '';
```

Comment:
The check for the existence of `fromSeparator` and `toSeparator` is also part of input validation and can be removed according to the task's instructions.